BrowserKit/Sources/SiteImageView/ImageProcessing/ImageHandler.swift

```diff
@@ -51,61 +49,59 @@ class DefaultImageHandler: ImageHandler {
         } catch {
-            return try await fetchHeroImageFromFetcher(site: site)
+            return try await fetchHeroImageFromFetcher(imageModel: imageModel)
         }
     }
 
     func clearCache() {
         Task {
-            await self.imageCache.clearCache()
+            await self.imageCache.clear()
         }
     }
 
     // MARK: Private
 
-    private func fetchFaviconFromFetcher(site: SiteImageModel) async -> UIImage {
+    private func fetchFaviconFromFetcher(imageModel: SiteImageModel) async -> UIImage {
         do {
-            guard let url = site.faviconURL else {
-                return await fallbackToLetterFavicon(site: site)
+            guard let faviconURL = imageModel.resourceURL else {
+                throw SiteImageError.noFaviconURLFound
             }
-
-            let image = try await faviconFetcher.fetchFavicon(from: url)
-            await imageCache.cacheImage(image: image, cacheKey: site.cacheKey, type: site.expectedImageType)
+            let image = try await faviconFetcher.fetchFavicon(from: faviconURL)
+            await imageCache.cacheImage(image: image, cacheKey: imageModel.cacheKey, type: imageModel.imageType)
             return image
         } catch {
-            return await fallbackToLetterFavicon(site: site)
+            return await fallbackToLetterFavicon(imageModel: imageModel)
         }
     }
 
-    private func fetchHeroImageFromFetcher(site: SiteImageModel) async throws -> UIImage {
-        guard let siteURL = site.siteURL else {
-            throw SiteImageError.noHeroImage
-        }
+    private func fetchHeroImageFromFetcher(imageModel: SiteImageModel) async throws -> UIImage {
         do {
-            let image = try await heroImageFetcher.fetchHeroImage(from: siteURL)
-            await imageCache.cacheImage(image: image, cacheKey: site.cacheKey, type: .heroImage)
+            let image = try await heroImageFetcher.fetchHeroImage(from: imageModel.siteURL)
+            await imageCache.cacheImage(image: image, cacheKey: imageModel.cacheKey, type: .heroImage)
             return image
         } catch {
             throw SiteImageError.noHeroImage
         }
     }
 
-    private func fallbackToLetterFavicon(site: SiteImageModel) async -> UIImage {
+    private func fallbackToLetterFavicon(imageModel: SiteImageModel) async -> UIImage {
         do {
-            let image = try await letterImageGenerator.generateLetterImage(siteString: site.cacheKey)
-            await imageCache.cacheImage(image: image, cacheKey: site.cacheKey, type: site.expectedImageType)
+            let image = try await letterImageGenerator.generateLetterImage(siteString: imageModel.cacheKey)
+            // FIXME Do we really want to cache letter icons and never attempt to get a favicon again?
+            //       We can drop into here on a network timeout.
+            await imageCache.cacheImage(image: image, cacheKey: imageModel.cacheKey, type: imageModel.imageType)
```

Comment:
We may want to make this caching logic a bit more sophisticated in the future. Even if there's a network timeout, we'll generate and save a letter favicon.

Reply:
This is a tricky one because the ideal is that we get the proper favicon at the next available opportunity but the mechanism for trying to get it is expensive and the most common case where a letter image is used is cases where a letter image will always be used so regenerating it often is also expensive. The middle ground we came up with was to have the url cache and the image cache expire cache entries after some time but this isn't a perfect solution because if a favicon is one that is used often by the user and it has defaulted to a letter image, the expiration will be reset regularly so it will never be fetched again.
I see letter image favicons more often on desktop than I do on mobile so I think we are in an acceptable spot for how we are handling it at the moment, but if you have an idea for how to better solve this then feel free to tackle it.

Reply:
The cache expiry is a good middle ground I think! It makes sense we'd want to cache the letter icon if generating it is expensive.

I don't have a quick way to refactor this off the top of my hand--I think it would involve a little re-architecting how the existing pieces fit together--but I'll point it out to Andy. There's potential for more favicon-related refactors but I think I'm out of time on this task since the Settings Redesign will be kicking off shortly. 😅 

I also just noticed the hero images also trigger multiple image fetches on cell refreshes just like the favicons (with regards to my [other PR](https://github.com/mozilla-mobile/firefox-ios/pull/21587)). They're just a bit less egregious since I think they require adding a bookmark generally, not just browsing pages.